### PR TITLE
.github: get build and CI triggered by opened, synchronize

### PR DESCRIPTION
**Pull request by dongsupark**
Build and CI tests should run automatically whenever a pull request is opened, reopened or updated.
On the other hand, it is not necessary to run build and CI tests on the events `ready_for_review` and `review_requested`.

That could already work out of the box for `reopened` and `synchronized` events.

To do that especially for the `opened` event, however, it is necessary to use a dedicated personal access token `BOT_PR_TOKEN` for bot PRs instead of the default `GITHUB_TOKEN`. It is needed for triggering another workflow from pull requests created by Github Actions. The default `GITHUB_TOKEN` is by design not able to trigger another workflow.

See also https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow.

## Testing done

tested via my fork repo

### CODE REVIEW

**Review by krnowak — APPROVED** [review]
Is the BOT_PR_TOKEN stuff stored somewhere in any secrets repo?

File: .github/workflows/update-portage-stable-packages-from-list.yaml
Comment (krnowak) [inline .github/workflows/update-portage-stable-packages-from-list.yaml:1]: I'm not sure if we want this action to trigger a build yet. Usually the generated PRs need more work, so running CI for them just after filing them is going to be a waste of cycles.

Comment (dongsupark) [reply .github/workflows/update-portage-stable-packages-from-list.yaml:1]: Right, dropped changes in update-portage...yaml.

**Comment by dongsupark** [comment]
> Is the BOT_PR_TOKEN stuff stored somewhere in any secrets repo?

Sent a DM.

**Pull request merged**